Exceptions/CheckoutExceptionLogger.php

```diff
@@ -0,0 +1,36 @@
+<?php
+declare(strict_types=1);
+
+namespace Paytrail\PaymentService\Exceptions;
+
+use Paytrail\PaymentService\Logger\PaytrailLogger;
+
+/**
+ * Class CheckoutExceptionLogger
+ */
+class CheckoutExceptionLogger
```

Comment:
generally looks fine, maybe the name of the class is not exactly self describing, as this is not only logger, but it throws exception after logging the error, which is maybe more important overall